File: Labs/Lab11/TowersOfHanoi.java
import java.util.Scanner;

public class TowersOfHanoi {

	public static void moves(int n, boolean leftMove) {
			
		if (n == 0) {				// base case	
			return;
		}
		moves(n-1, !leftMove);
		if (leftMove) {
			System.out.println( n + " left ");				// write (OS), printf, writeStr
		} else {
			System.out.println( n + " right ");
		}
		moves(n-1, !leftMove);
	}
	
	public static void main(String[] args) {
		int n;
		
		Scanner myInput = new Scanner( System.in );
	    System.out.print( "Enter number of game disks: " );
	    n = myInput.nextInt();
	    myInput.close();
	
		moves(n, true);
		System.out.println("done");
	}	
}

File: Labs/Lab11/TowersOfHanoi.c
#include <stdio.h>
#include <stdbool.h>

void moves(int n, bool leftMove)
{
    if(n == 0){
        return;
    }
    moves(n-1, !leftMove);
    // if(leftMove) {
    //     printf("%d", n);
    //     printf("%s", " left \n");
    // }
    // else{
    //     printf("%d", n);
    //     printf("%s", " right \n");
    // }
    moves(n-1, !leftMove);
}

int main()
{
    int n = 3;

    // printf("%s", "Enter number of game disks: ");
    // scanf("%d", &n);
    moves(n, true);
    // printf("%s", "done\n");

    return 0;
}